Search status tab

DoumanAsh opened the pull request:
#2859 leads to idea that it would be good to align Search status with current tab.
Draft commit for that

sledgehammer999 commented:
Can you rebase and maybe post a screenshot?

DoumanAsh commented:
Yes, i will. Hope it will be ok...

DoumanAsh commented:
Rebased, but i suspect that i may need to re-make it from a scratch. Will see that from build results. (i deleted my VM unix, so i'm not able to build by myself right now)

ngosang commented:
I don't understand well what this PR does.

DoumanAsh commented:
@ngosang Well i think i wanted to have separate state machine for currently Active search tab.
I was doing it quite some time ago and well i don't really remember if i implemented everything. On weekends i will setup back my unix VM and will see my commit in action

sledgehammer999 commented:
Ok, once you're satisfied with your code, tag me again.

DoumanAsh commented:
@sledgehammer999 
Can you please review me?

I changed handling of tab closing, now instead of stopping search when you close current tab it is checking for active tab(where is search actually ongoing)

DoumanAsh commented:
Yesterday i tried to build qBittorrent with msys2.
Everything is fine except for Boost :( For some reason ./configure does not see boost libs.
Will try to work it out, if i'll be lucky i'll try to make new guide for build with msys2

sledgehammer999 commented:
>Yesterday i tried to build qBittorrent with msys2.

I have done it in the past but switched to MSVC for smaller binaries.

>Everything is fine except for Boost :( For some reason ./configure does not see boost libs.

Use the b2.exe or bjam.exe to compile boost and libtorrent. They get created when you bootstrap boost(another script file in boost sources). Just be sure to launch them from inside msys.

Sorry to ask you again, but can you rebase?

sledgehammer999 commented:
From this forum post onwards I describe what I did with gcc 4.7.x to compile qbt and relevant libs: https://qbforums.shiki.hu/index.php/topic,1000.msg4039.html#msg4039

DoumanAsh commented:
Oo... I totally forgot about this PR :D i was busy recently with preparations to move to new job and learning Rust...

I will rebase and will try to follow your description

sledgehammer999 commented:
I didn't fully test the rest of the changes due to my comments and one thing I observed:
Let's say you initiate a search and while it is searching you initiate a second one. The status remains as "Searching..." which is correct.
However while the second search is running if you change to the first tab the status goes to "Stopped" which is also correct. But changing back to the second tab which is still searching it doesn't change the Status text to "Searching..." it continues to display as "Stopped"

DoumanAsh commented:
Thanks for small test and review.
I'll see to it

DoumanAsh commented:
Updated my commit.
I also checked all places where ```currentSearchTab``` is used and replaced it with ```activeSearchTab``` when it comes to handling search results. at least i hope i was doing that... :)

sledgehammer999 commented on src/searchengine/searchengine.cpp:
```diff
@@ -409,6 +414,8 @@ void SearchEngine::updateNova() {
 }
 
+#define SET_SEARCH_STATUS(text) \
+    activeSearchTab->search_status = tr(text);
```
This macro is totally useless. Just use `activeSearchTab->search_status = tr(text);` directly.

sledgehammer999 commented on src/searchengine/searchengine.cpp:
```diff
@@ -156,6 +157,7 @@ void SearchEngine::tab_changed(int t)
             download_button->setEnabled(false);
             goToDescBtn->setEnabled(false);
         }
+        search_status->setText(currentSearchTab->search_status);
```
you need a QString here but you pass a QLabel (hence the travis error)

DoumanAsh commented:
Sorry, everything about search engine code is confusing :)
I'll need to think about how to design it properly...

sledgehammer999 commented:
This seems to work fine and the code looks ok. Thank you.

DoumanAsh commented:
That's good. Thanks for verification :)
